coderd/userauth.go

```diff
@@ -1343,13 +1343,12 @@ func (api *API) oauthLogin(r *http.Request, params *oauthLoginParams) ([]*http.C
 		if user.ID == uuid.Nil {
 			var organizationID uuid.UUID
 			//nolint:gocritic
-			organizations, _ := tx.GetOrganizations(dbauthz.AsSystemRestricted(ctx))
-			if len(organizations) > 0 {
-				// Add the user to the first organization. Once multi-organization
-				// support is added, we should enable a configuration map of user
-				// email to organization.
-				organizationID = organizations[0].ID
-			}
+			organization, _ := tx.GetDefaultOrganization(dbauthz.AsSystemRestricted(ctx))
+
+			// Add the user to the default organization.
+			// Once multi-organization we should check some configuration to see
+			// if we should add the user to a different organization.
+			organizationID = organization.ID
```

Comment:
This is annoyingly allowed to be `uuid.Nil` in unit tests. I am not going to change the behavior here, but we should really fix that so we don't have an edge case living for the sake of tests.

Reply:
Fine not to fix here, but can we add a comment for the ignored err above (`tx.GetDefaultOrganization`) about this? Comment is just a band-aid though, we definitely want to nip this in the bud before too long.